Extension/src/LanguageServer/configurations.ts

```diff
@@ -117,32 +122,7 @@ export class CppProperties {
         let rootPath: string = rootUri ? rootUri.fsPath : "";
         this.currentConfigurationIndex = new PersistentFolderState<number>("CppProperties.currentConfigurationIndex", -1, rootPath);
         this.configFolder = path.join(rootPath, ".vscode");
-
-        let configFilePath: string = path.join(this.configFolder, "c_cpp_properties.json");
-        if (fs.existsSync(configFilePath)) {
-            this.propertiesFile = vscode.Uri.file(configFilePath);
-            this.parsePropertiesFile();
-        }
-        if (!this.configurationJson) {
-            this.resetToDefaultSettings(this.CurrentConfigurationIndex === -1);
-        }
-
-        this.configFileWatcher = vscode.workspace.createFileSystemWatcher(path.join(this.configFolder, this.configurationGlobPattern));
-        this.disposables.push(this.configFileWatcher);
-        this.configFileWatcher.onDidCreate((uri) => {
-            this.propertiesFile = uri;
-            this.handleConfigurationChange();
-        });
-
-        this.configFileWatcher.onDidDelete(() => {
-            this.propertiesFile = null;
-            this.resetToDefaultSettings(true);
-            this.handleConfigurationChange();
-        });
-
-        this.configFileWatcher.onDidChange(() => {
-            this.handleConfigurationChange();
-        });
+        this.diagnosticCollection = vscode.languages.createDiagnosticCollection(rootPath);
```

Comment:
is this object supposed to be disposed at some point?

Reply:
and is it supposed to be for the c_cpp_properties.json file instead of the workspace folder?

Reply:
Not sure, I'll check...

Reply:
If it's disposable, we just need to add it to the disposables array

Reply:
This seems correct to me. It doesn't need to be disposed. Using rootPath as the "name" should be equivalent to the rootPath + .vscode/c_cpp_properties.json and that last part is unnecessary, right?

Reply:
I just looked at the API.  The string you pass in is just a name for the collection, not a uri or filepath.  And it does have a dispose method on it, so you should call `dispose` on it in the `dispose` method at the bottom of this file.  It doesn't look like it inherits `vscode.Disposable` so you can't add it to the disposables array.

Reply:
I added a dispose call.